### PR TITLE
[ZEPPELIN-4321] Rework Zeppelin with Latest Angular

### PR DESCRIPTION
**Pull request by hsuanxyz**
### What is this PR for?

The old version of zeppelin was written with AngularJS, there exist many issues related to the zeppelin-web and could not resolve due to the low version of AngularJS.

So we rework the zeppelin-web with the latest Angular (8.0).

### What type of PR is it?

Feature

#### Pages

| Name | Route | Module     | UI |
| ---  | ----- | ---------- | -- |
| Home | `/`   | HomeModule |  Y |
| Login | `/login`       | LoginModule |  Y |
| Job Manager  | `/jobmanager`  | JobManagerModule |  Y |
| Interpreter Setting   | `/interpreter` | InterpreterModule |  Y |
| Notebook | `/notebook/{id}` | NotebookModule |  Y |
| Notebook Repos | `/notebookRepos` |  |   |
| Credential     | `/credential` |  |   |
| Helium | `/helium` |  |  WIP  |
| Configuration | `/configuration` |  |   |

#### Notebook Features

| Feature | Description | Status |
| ------  | ---- | ---- |
| Files System  | Create/ Rename/ Import etc.  | Y |
| Toolbar Actions  | The top toolbar actions   | Y |

#### Paragraph Features

| Feature | Description | Status |
| ------  | ---- | ---- |
| Grid layout and resizable | | Y |
| Code Editor | | Y |
| Actions  | The Corresponding actions of the drop-down menu in the setting button | Y |
| Actions(hot-keys)  | Support hot-keys for the actions  | WIP |
| Publishable  | [publish paragraphs](http://zeppelin.apache.org/docs/0.8.0/usage/other_features/publishing_paragraphs.html)  |  |
| Stream  |  |  |

#### Result Display

| Type | Status |
| ------  | ---- |
| Dynamic Form  | Y |
| Text  | Y |
| Html  |  Y |
| Table  |  Y |
| Network  |  |

#### Table Visualization

| Type | State |
| ------ | ---- |
| Line Chart  | Y |
| Bard Chart  |  Y |
| Pie Chart  |  Y |
| Area Chart  |  Y |
| Scatter Chart  | Y |

#### Helium Visualization

| Type | Description | Status |
| ------  | ---- | ---- |
| Prototype | To verify the implementable prototype | Y |
| Publish Dependencies | Just like [zeppelin-vis](https://github.com/apache/zeppelin/tree/master/zeppelin-web/src/app/visualization)  | WIP |
| Example Projects |   | Y |
| Development Documents |   | WIP |

### What is the Jira issue?

https://issues.apache.org/jira/browse/ZEPPELIN-4321
https://issues.apache.org/jira/browse/ZEPPELIN-4215

#### Prerequisites

- [Node.js](https://nodejs.org) version 10.9.0 or later or use [creationix/nvm](https://github.com/creationix/nvm).
- NPM package manager (which is installed with Node.js by default).
- [Angular CLI](https://angular.io/cli) version 8.3.0 or later.

#### Install

```bash
$ cd zeppelin-frontend
$ npm install
```
Install dependencies in the project directory.

### Start Zeppelin server

[Run Zeppelin server](https://zeppelin.apache.org/contribution/contributions.html#run-zeppelin-server-in-development-mode) on `http://localhost:8080`.

If you are using a custom port instead of the default(http://localhost:8080) or other network address, you can create `.env` file in the project directory and set `SERVER_PROXY`.

*.env*

```
SERVER_PROXY=http://localhost:8080
```

### Development server

```bash
$ npm start
```
Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.

### Build

Run `npm build` to build the project. The build artifacts will be stored in the `dist/` directory.

### Screenshots (if appropriate)

**Login**

login

**Home**

image
image

**Interpreter**

zp-interpreter

**Job Manager**

zp-jm

**Dynamic Form**

zp-df

**Frontend API(latest Angular)**

zp-ng

** Visualization**

zp-vis

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? Maybe
* Does this needs documentation? YES

### CODE REVIEW

**Comment by zjffdu** [comment]
@hsuanxyz Could you add apache license header ?

**Comment by hsuanxyz** [comment]
@zjffdu Ok, I will add a tslint rule that will automatically be added the license header when each time commit.

**Comment by zjffdu** [comment]
Thanks @hsuanxyz In the PR description, the `Note` column in table should be renamed to `Description`

File: zeppelin-frontend/README.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+### Technologies
+
+Zeppelin-Frontend-Next is using Angular as the main Framework, before developing we hope highly recommended to have a good knowledge of [Angular](https://angular.io/) and [RxJs](https://github.com/ReactiveX/rxjs).
+
+In addition:
+
+- We use [G2](https://github.com/antvis/g2) visualization
+- We use [Lodash](https://lodash.com/) to process complex data
+- We use [Monaco Editor](https://github.com/microsoft/monaco-editor) to make code editor
```
Comment (zjffdu) [inline zeppelin-frontend/README.md]: What are licences of these dependencies ? We need to update `LICENCE` file accordingly.

**Comment by conker84** [comment]
@hsuanxyz why is the network visualization not implemented?

**Comment by hsuanxyz** [comment]
@conker84 We plan to continue development after this PR review is done. All tasks in the list will be completed.

File: zeppelin-frontend/README.md (hunk 1 above)
Comment (hsuanxyz) [reply zeppelin-frontend/README.md]: Done

**Comment by hsuanxyz** [comment]
@zjffdu 

> 1. In the PR description, the Note column in table should be renamed to Description

- Done

> 2. Does package-lock.json need to be committed ?

- Added

I have added the `pom.xml` configuration file, now `build` and `test` will run in CI. We will add more tests after the project is relatively stable.

**Comment by zjffdu** [comment]
I have merged it, could you close this PR ? @hsuanxyz

**Comment by hsuanxyz** [comment]
> I have merged it, could you close this PR ?

@zjffdu Thank your.

**Pull request closed without merging**